### DISCUSSION TITLE
PYBv1.1 Pyboard (2016) can't connect to Windows 11 2022 DELL PC

### DISCUSSION THREAD

**Discussion by jerrylow2000** [pyControl/code#110] [Q&A]
Hi I'm very new for Pyboard. I tried to use the GUI to connect the board but somehow it couldn't. The error log is following and could anyone answer my problem, thank you. 

2023-12-12 18:30:44,758 Traceback (most recent call last):
  File "C:\Users\kail\Downloads\pyControl_v1.8.1_TEST\gui\run_task_tab.py", line 71, in <lambda>
    self.connect_button.clicked.connect(lambda: self.disconnect() if self.connected else self.connect())
                                                                                         ^^^^^^^^^^^^^^
  File "C:\Users\kail\Downloads\pyControl_v1.8.1_TEST\gui\run_task_tab.py", line 257, in connect
    self.board = Pycboard(port, print_func=self.print_to_log, data_logger=self.data_logger)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kail\Downloads\pyControl_v1.8.1_TEST\com\pycboard.py", line 71, in __init__
    self.unique_ID = eval(self.eval('pyb.unique_id()').decode())
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 0
    
SyntaxError: invalid syntax

**Comment by ThomasAkam** [comment]
Hi @jerrylow2000 

I'm not sure what is causing this problem but the first things to try if you can't connect to a board are:

1.  Close the GUI, press the reset buton on the pyboard (or on the pycontrol breakout board if your using one) then reopen the GUI and try to connect again.
2. If that does not work, try resetting the pyboard filesystem by following the instructions [here](https://docs.micropython.org/en/latest/pyboard/tutorial/reset.html#factory-reset-the-filesystem), then try again to connect.

**Reply by terezie-sedlinska** [reply]
Dear all, we have the same problem after switching to a new PC. We have tried both resets and it hasn’t worked. Is there another trick we could try?

**Reply by ThomasAkam** [reply]
Hi @terezie-sedlinska, can you provide a bit more information about the issue.  
- When exactly does the error happen?  Does the board show up in the GUI but you are unable to connect, or does it not show up in the GUI at all?  
- Does the board work ok on the old computer?  
- Are you using a different OS on the two computers?  Is the computer OS langauge set to a language other than english?
- Do you see anything in the ErrorLog.txt file in the pycontrol folder?

**Comment by alustig3** [comment]
@jerrylow2000 it may also be worth confirming that your cable is usb power _and_ data. Some usb cables are power only. In windows search for and open "Device manager" and click the "Ports" toggle. When you plug in your pyboard, a new USB Serial Device should appear.

<img width="325" alt="CleanShot 2023-12-13 at 12 02 49@2x" src="https://github.com/pyControl/code/assets/8128628/e038b438-d5a8-45b6-a52b-c04493ba168d">

**Comment by terezie-sedlinska** [comment]
Dear Thomas,

Thank you for your quick reply.

To answer your questions:

  1.  The board doesn’t show up in the GUI.
  2.  Yes, it has worked until now and after switching it back to the old computer it worked again.
  3.  Both computers are windows. The old one is in Japanese (but there the connection works). The new one is in English.
  4.  I can have a look when I’m in the lab tomorrow.

Thank you and all the best,

Terezie

PhD student
Neural Computation Unit (Doya Unit)
Okinawa Institute of Science and Technology


From: Thomas Akam ***@***.***>
Date: Friday, July 19, 2024 at 19:53
To: pyControl/code ***@***.***>
Cc: Terezie Sedlinska ***@***.***>, Mention ***@***.***>
Subject: Re: [pyControl/code] PYBv1.1 Pyboard (2016) can't connect to Windows 11 2022 DELL PC (Discussion #110)

Hi @terezie-sedlinska<https://github.com/terezie-sedlinska>, can you provide a bit more information about the issue.

  *   When exactly does the error happen? Does the board show up in the GUI but you are unable to connect, or does it not show up in the GUI at all?
  *   Does the board work ok on the old computer?
  *   Are you using a different OS on the two computers? Is the computer OS langauge set to a language other than english?
  *   Do you see anything in the ErrorLog.txt file in the pycontrol folder?

**Reply by ThomasAkam** [reply]
@terezie-sedlinska could you try opening the gui then unplugging and re-plugging in the pyboard to the computer to see if it then shows up in the gui.  We have sometimes had an issue where the boards don't show up when the gui is first opened but then appear when unplugged and plugged back in.
